test/unit/marked-spec.js

```diff
@@ -137,8 +137,89 @@ describe('parseInline', () => {
 });
 
 describe('use extension', () => {
+  it('should use custom block tokenizer + renderer extensions', () => {
+    const underline = {
+      name: 'underline',
+      before: 'paragraph', // Leave blank to run after everything else...?
+      level: 'block',
+      tokenizer: (src) => {
+        const rule = /^:([^\n]*)(?:\n|$)/;
+        const match = rule.exec(src);
+        if (match) {
+          return {
+            type: 'underline',
+            raw: match[0], // This is the text that you want your token to consume from the source
+            text: match[1].trim() // You can add additional properties to your tokens to pass along to the renderer
+          };
+        }
+      },
+      renderer: (token) => {
+        return `<u>${token.text}</u>\n`;
+      }
+    };
+    marked.use(underline);
+    let html = marked('Not Underlined\n:Underlined\nNot Underlined');
+    expect(html).toBe('<p>Not Underlined\n:Underlined\nNot Underlined</p>\n');
+
+    html = marked('Not Underlined\n\n:Underlined\n\nNot Underlined');
+    expect(html).toBe('<p>Not Underlined</p>\n<u>Underlined</u>\n<p>Not Underlined</p>\n');
+  });
+
+  it('should interrupt paragraphs if using "start" property', () => {
+    const underline = {
+      name: 'underline',
+      before: 'paragraph', // Leave blank to run after everything else...?
+      level: 'block',
+      start: /:/,
+      tokenizer: (src) => {
+        const rule = /^:([^\n]*)(?:\n|$)/;
+        const match = rule.exec(src);
+        if (match) {
+          return {
+            type: 'underline',
+            raw: match[0], // This is the text that you want your token to consume from the source
+            text: match[1].trim() // You can add additional properties to your tokens to pass along to the renderer
+          };
+        }
+      },
+      renderer: (token) => {
+        return `<u>${token.text}</u>\n`;
+      }
+    };
+    marked.use(underline);
+    const html = marked('Not Underlined\n:Underlined\nNot Underlined');
+    expect(html).toBe('<p>Not Underlined</p>\n<u>Underlined</u>\n<p>Not Underlined</p>\n');
+  });
+
+  it('should use custom inline tokenizer + renderer extensions', () => {
+    const underline = {
+      name: 'underline',
+      before: 'emStrong', // Leave blank to run after everything else...?
+      level: 'inline',
+      start: /=/,
+      tokenizer: (src) => {
+        const rule = /^=([^=]+)=/;
+        const match = rule.exec(src);
+        if (match) {
+          return {
+            type: 'underline',
+            raw: match[0], // This is the text that you want your token to consume from the source
+            text: match[1].trim() // You can add additional properties to your tokens to pass along to the renderer
+          };
+        }
+      },
+      renderer: (token) => {
+        return `<u>${token.text}</u>`;
+      }
+    };
+    marked.use(underline);
```

Comment:
The extension should be added as an extension option.

```js
marked.use({
	extension: underline
	extension: [underline] // or as an array
});
```

That way it won't be a breaking change for `marked.use`,

Reply:
> The only breaking change was adding an overwrite flag to existing extensions

So this is a breaking change currently.